core/src/test/java/org/apache/iceberg/rest/responses/TestLoadTableResponse.java

```diff
@@ -0,0 +1,371 @@
+        "      \"type\":\"tag\"}," +
+        "    \"test\":{" +
+        "      \"snapshot-id\":" + metaV2.ref("test").snapshotId() + "," +
+        "     \"type\":\"branch\"}" +
+        "  }," +
+        "  \"snapshots\":[{" +
+        "    \"snapshot-id\":" + metaV2.snapshots().get(0).snapshotId() + "," +
+        "    \"timestamp-ms\":" + metaV2.snapshots().get(0).timestampMillis() + "," +
+        "    \"manifests\":[" +
+        "      \"/tmp/manifest.1.avro\"" +
+        "    ]},{" +
+        "    \"snapshot-id\":" + metaV2.snapshots().get(1).snapshotId() + "," +
+        "    \"parent-snapshot-id\":" + metaV2.snapshots().get(1).parentId() + "," +
+        "    \"timestamp-ms\":" + metaV2.snapshots().get(1).timestampMillis() + "," +
+        "    \"manifests\":[" +
+        "      \"/tmp/manifest.2.avro\"" +
+        "    ],\"schema-id\":7" +
+        "  }]," +
+        "  \"snapshot-log\":[{" +
+        "      \"timestamp-ms\":" + previousSnapshot.timestampMillis() + "," +
+        "      \"snapshot-id\":" + previousSnapshot.snapshotId() +
+        "    },{" +
+        "      \"timestamp-ms\":" + currentSnapshot.timestampMillis() + "," +
+        "      \"snapshot-id\":" + currentSnapshot.snapshotId() +
+        "  }]," +
+        "  \"metadata-log\":[]}," +
+        "  \"config\":{\"foo\":\"bar\"}}";
+
+    assertRoundTripSerializesEquallyFrom(fullExampleInstanceJson.replaceAll("\\s", ""), respV2);
+
+  }
+
+
+  private TableMetadata buildV2TableMetadataWithTagsAndSnapshotHistory() {
+    Snapshot previousSnapshot = new BaseSnapshot(
+        ops.io(), PREVIOUS_SNAPSHOT_ID, null, PREVIOUS_SNAPSHOT_ID, null, null, null, ImmutableList.of(
+            new GenericManifestFile(localInput("file:/tmp/manifest.1.avro"), SPEC_5.specId())));
+    long currentSnapshotId = System.currentTimeMillis();
+    Snapshot currentSnapshot = new BaseSnapshot(
+        ops.io(), currentSnapshotId, PREVIOUS_SNAPSHOT_ID, currentSnapshotId, null, null, 7, ImmutableList.of(
+            new GenericManifestFile(localInput("file:/tmp/manifest.2.avro"), SPEC_5.specId())));
+
+    List<HistoryEntry> snapshotLog = ImmutableList.<HistoryEntry>builder()
+        .add(new TableMetadata.SnapshotLogEntry(previousSnapshot.timestampMillis(), previousSnapshot.snapshotId()))
+        .add(new TableMetadata.SnapshotLogEntry(currentSnapshot.timestampMillis(), currentSnapshot.snapshotId()))
+        .build();
+
+    Schema schema = new Schema(6,
+        Types.NestedField.required(10, "x", Types.StringType.get()));
+
+    Map<String, SnapshotRef> refs = ImmutableMap.of(
+        "main", SnapshotRef.branchBuilder(currentSnapshotId).build(),
+        "previous", SnapshotRef.tagBuilder(PREVIOUS_SNAPSHOT_ID).build(),
+        "test", SnapshotRef.branchBuilder(PREVIOUS_SNAPSHOT_ID).build()
+    );
+
+    // TODO - Use builder to get here instead - possibly even just writing out to a .json file and reading in from
+    //  there.
```

Comment:
I agree with this. You can use `TableMetadata.buildFrom` and `TableMetadata.newTableMetadata` instead of making the constructor public.